cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go

```diff
@@ -105,18 +109,39 @@ func (ihp *incomingHeaderProcessor) AddHeader(headerHash []byte, header sovereig
 		return err
 	}
 
-	ihp.eventsProc.addSCRsToPool(incomingSCRs)
+	ihp.addConfirmedBridgeOpsToPool(res.confirmedBridgeOps)
+	ihp.addSCRsToPool(res.scrs)
 	return nil
 }
 
+func (ihp *incomingHeaderProcessor) addSCRsToPool(scrs []*scrInfo) {
+	cacheID := process.ShardCacherIdentifier(core.MainChainShardId, core.SovereignChainShardId)
+
+	for _, scrData := range scrs {
+		ihp.txPool.AddData(scrData.hash, scrData.scr, scrData.scr.Size(), cacheID)
+	}
+}
+
+func (ihp *incomingHeaderProcessor) addConfirmedBridgeOpsToPool(ops []*confirmedBridgeOp) {
+	for _, op := range ops {
+		// This is not a critical error. This might just happen when a leader tries to re-send unconfirmed confirmation
+		// that have been already executed, but the confirmation from notifier comes too late, and we receive a double
+		// confirmation.
+		err := ihp.outGoingPool.ConfirmOperation(op.hashOfHashes, op.hash)
+		if err != nil {
+			log.Debug("incomingHeaderProcessor.AddHeader.addConfirmedBridgeOpsToPool", "error", err)
```

Comment:
Perhaps it would be helpful if the log also included the hash of the operation.

Reply:
Returned error contains the hash of the operation already, but can add it again just to be extra sure